File: app_java/src/main/java/vlab/android/architecture/di/module/ApiModule.java
package vlab.android.architecture.di.module;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;
import vlab.android.architecture.data.source.remote.GitHubApi;

/**
 * Created by Vinh Tran on 2/11/18.
 */
@Module
public class ApiModule {

    @Singleton
    @Provides
    GitHubApi provideGitHubApi(Retrofit retrofit){
        return retrofit.create(GitHubApi.class);
    }

}

File: app_kotlin/src/main/java/vlab/android/architecture/di/module/ApiModule.kt
package vlab.android.architecture.di.module

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import vlab.android.architecture.repository.source.remote.GitHubApi
import javax.inject.Singleton

/**
 * Created by vinhtran on 2/11/18.
 */
@Module
class ApiModule {

    @Singleton
    @Provides
    internal fun provideGitHubApi(retrofit: Retrofit): GitHubApi {
        return retrofit.create(GitHubApi::class.java)
    }

}
